bigtable-hbase-1.x-parent/bigtable-hbase-1.x/src/main/java/com/google/cloud/bigtable/hbase1_x/BigtableConnection.java

```diff
@@ -58,7 +65,28 @@ public BigtableConnection(Configuration conf) throws IOException {
   /** {@inheritDoc} */
   @Override
   public Admin getAdmin() throws IOException {
-    return new BigtableAdmin(this);
+    try {
+      BigtableAdmin admin =
+          new ByteBuddy()
+              .subclass(BigtableAdmin.class)
+              .defineConstructor(Visibility.PUBLIC)
+              .intercept(
+                  MethodCall.invoke(
+                          BigtableAdmin.class.getDeclaredConstructor(CommonConnection.class))
+                      .with(this))
+              .method(ElementMatchers.isAbstract())
+              .intercept(
+                  InvocationHandlerAdapter.of(
+                      new AbstractBigtableAdmin.UnsupportedOperationsHandler()))
+              .make()
+              .load(BigtableAdmin.class.getClassLoader())
+              .getLoaded()
```

Comment:
I dont think we should be defining a new class for every invocation. Instead I think it would be better to move the class definition to BigtableAdmin, something along the lines of:

abstract class BIgtableAdmin {
  private static Class<? extends BigtableAdmin> classDefinition = null;
  private static synchronized  Class<? extends BigtableAdmin> getSubclass() {
    if (classDefinition == null) {
      // byte buddy...
    }
    return classDefinition;
  }
   public static Admin createInstance(Connection connection) {
      return getSubclass().newInstance()...;
   }